[HELP] Server Error (500)

**Describe the bug**
When signing in PostHog after installation, `Server Error (500)` occurred.


**To Reproduce**
1. access PostHog
2. trying to login

**Expected behavior**
Login successfully.

**Screenshots**
None.

**Server:**
 - OS: Debian 10.3
 - Proxy: Apache 2.4.38
 - Python: 3.7.7 
 - PostHog Version: latest of `master` branch

**gunicorn start script**
```sh
#! /bin/sh

PORT=9099
LOGLEVEL=debug
WORKERS=2
THREADS=5

HOME=/srv/posthog
cd $HOME

if [ -d "$HOME/.pyenv/bin" ] ; then
    PATH="$HOME/.pyenv/bin:$PATH"
    eval "$(pyenv init -)"
fi

DATABASE_URL="postgres://posthog:password@192.168.207.11:5432/posthog?sslmode=disable"
REDIS_URL="redis://192.168.207.1:6379"
export DATABASE_URL REDIS_URL

EMAIL_HOST="192.168.207.21"
EMAIL_PORT="25"
DEFAULT_FROM_EMAIL="no-reply@nueh.net"
export EMAIL_HOST EMAIL_PORT DEFAULT_FROM_EMAIL

IS_BEHIND_PROXY=true
DISABLE_SECURE_SSL_REDIRECT=true
export IS_BEHIND_PROXY DISABLE_SECURE_SSL_REDIRECT

LOGDIR="$HOME/logs"
RUNDIR="$HOME/run"

OPTS=
OPTS=$OPTS" --chdir "$HOME"/posthog"
OPTS=$OPTS" --config posthog/gunicorn.config.py"
OPTS=$OPTS" --access-logfile "$LOGDIR"/access.log"
OPTS=$OPTS" --error-logfile "$LOGDIR"/error.log"
OPTS=$OPTS" --log-level "$LOGLEVEL
OPTS=$OPTS" --pid "$RUNDIR"/posthog.pid"
OPTS=$OPTS" --bind 127.0.0.1:"$PORT
OPTS=$OPTS" --workers "$WORKERS
OPTS=$OPTS" --threads "$THREADS

exec gunicorn $OPTS posthog.wsgi
```

**gunicorn access log**
```
127.0.0.1 - - [11/Apr/2020:14:03:01 +0000] "GET / HTTP/1.1" 302 0 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.92 Safari/537.36"
127.0.0.1 - - [11/Apr/2020:14:03:01 +0000] "GET /login?next=/ HTTP/1.1" 200 3170 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.92 Safari/537.36"
127.0.0.1 - - [11/Apr/2020:14:03:01 +0000] "GET /static/main.css HTTP/1.1" 304 0 "https://p.nueh.net/login?next=/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.92 Safari/537.36"
127.0.0.1 - - [11/Apr/2020:14:03:08 +0000] "POST /login HTTP/1.1" 500 145 "https://p.nueh.net/login?next=/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.92 Safari/537.36"
127.0.0.1 - - [11/Apr/2020:14:03:08 +0000] "GET /favicon.ico HTTP/1.1" 302 0 "https://p.nueh.net/login" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.92 Safari/537.36"
127.0.0.1 - - [11/Apr/2020:14:03:08 +0000] "GET /login?next=/favicon.ico HTTP/1.1" 200 3170 "https://p.nueh.net/login" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.92 Safari/537.36"
```

**gunicorn error log**
```
[2020-04-11 07:02:33 -0700] [3433] [DEBUG] Current configuration:
  config: posthog/gunicorn.config.py
  bind: ['127.0.0.1:9099']
  backlog: 2048
  workers: 2
  worker_class: sync
  threads: 5
  worker_connections: 1000
  max_requests: 0
  max_requests_jitter: 0
  timeout: 30
  graceful_timeout: 30
  keepalive: 2
  limit_request_line: 4094
  limit_request_fields: 100
  limit_request_field_size: 8190
  reload: False
  reload_engine: auto
  reload_extra_files: []
  spew: False
  check_config: False
  preload_app: False
  sendfile: None
  reuse_port: False
  chdir: /srv/posthog/posthog
  daemon: False
  raw_env: []
  pidfile: /srv/posthog/run/posthog.pid
  worker_tmp_dir: None
  user: 1004
  group: 1004
  umask: 0
  initgroups: False
  tmp_upload_dir: None
  secure_scheme_headers: {'X-FORWARDED-PROTOCOL': 'ssl', 'X-FORWARDED-PROTO': 'https', 'X-FORWARDED-SSL': 'on'}
  forwarded_allow_ips: ['127.0.0.1']
  accesslog: /srv/posthog/logs/access.log
  disable_redirect_access_to_syslog: False
  access_log_format: %(h)s %(l)s %(u)s %(t)s "%(r)s" %(s)s %(b)s "%(f)s" "%(a)s"
  errorlog: /srv/posthog/logs/error.log
  loglevel: debug
  capture_output: False
  logger_class: gunicorn.glogging.Logger
  logconfig: None
  logconfig_dict: {}
  syslog_addr: udp://localhost:514
  syslog: False
  syslog_prefix: None
  syslog_facility: user
  enable_stdio_inheritance: False
  statsd_host: None
  dogstatsd_tags:
  statsd_prefix:
  proc_name: None
  default_proc_name: posthog.wsgi
  pythonpath: None
  paste: None
  on_starting: <function on_starting at 0x7f84616b33b0>
  on_reload: <function OnReload.on_reload at 0x7f84616145f0>
  when_ready: <function WhenReady.when_ready at 0x7f8461614710>
  pre_fork: <function Prefork.pre_fork at 0x7f8461614830>
  post_fork: <function Postfork.post_fork at 0x7f8461614950>
  post_worker_init: <function PostWorkerInit.post_worker_init at 0x7f8461614a70>
  worker_int: <function WorkerInt.worker_int at 0x7f8461614b90>
  worker_abort: <function WorkerAbort.worker_abort at 0x7f8461614cb0>
  pre_exec: <function PreExec.pre_exec at 0x7f8461614dd0>
  pre_request: <function PreRequest.pre_request at 0x7f8461614ef0>
  post_request: <function PostRequest.post_request at 0x7f8461614f80>
  child_exit: <function ChildExit.child_exit at 0x7f846162d0e0>
  worker_exit: <function WorkerExit.worker_exit at 0x7f846162d200>
  nworkers_changed: <function NumWorkersChanged.nworkers_changed at 0x7f846162d320>
  on_exit: <function OnExit.on_exit at 0x7f846162d440>
  proxy_protocol: False
  proxy_allow_ips: ['127.0.0.1']
  keyfile: None
  certfile: None
  ssl_version: 2
  cert_reqs: 0
  ca_certs: None
  suppress_ragged_eofs: True
  do_handshake_on_connect: False
  ciphers: None
  raw_paste_global_conf: []
  strip_header_spaces: False
[2020-04-11 07:02:33 -0700] [3433] [INFO] Starting gunicorn 20.0.4
[2020-04-11 07:02:33 -0700] [3433] [DEBUG] Arbiter booted
[2020-04-11 07:02:33 -0700] [3433] [INFO] Listening at: http://127.0.0.1:9099 (3433)
[2020-04-11 07:02:33 -0700] [3433] [INFO] Using worker: threads
[2020-04-11 07:02:33 -0700] [3548] [INFO] Booting worker with pid: 3548
[2020-04-11 07:02:33 -0700] [3549] [INFO] Booting worker with pid: 3549
[2020-04-11 07:02:33 -0700] [3433] [DEBUG] 2 workers
[2020-04-11 14:03:00 +0000] [3549] [DEBUG] GET /
[2020-04-11 14:03:01 +0000] [3549] [DEBUG] GET /login
[2020-04-11 14:03:01 +0000] [3549] [DEBUG] GET /static/main.css
[2020-04-11 14:03:07 +0000] [3548] [DEBUG] POST /login
[2020-04-11 14:03:08 +0000] [3548] [DEBUG] GET /favicon.ico
[2020-04-11 14:03:08 +0000] [3548] [DEBUG] GET /login
```

**apache config**
```apache
<IfModule mod_ssl.c>
<VirtualHost *:443>

	ServerName p.nueh.net
	ServerAdmin webmaster@nueh.net

	Include /etc/letsencrypt/options-ssl-apache.conf
	SSLCertificateFile /etc/letsencrypt/live/nueh.net/fullchain.pem
	SSLCertificateKeyFile /etc/letsencrypt/live/nueh.net/privkey.pem

	RequestHeader set X-Forwarded-Proto "https"
	ProxyPreserveHost On

	RewriteEngine On
	RewriteCond %{HTTP:Connection} Upgrade [NC]
	RewriteCond %{HTTP:Upgrade} websocket [NC]
	RewriteRule /(.*) ws://127.0.0.1:9099/$1 [P,L]

	ProxyPass / http://127.0.0.1:9099/
	ProxyPassReverse / http://127.0.0.1:9099/

	ErrorLog ${APACHE_LOG_DIR}/error.log
	CustomLog ${APACHE_LOG_DIR}/access.log combined

</VirtualHost>
</IfModule>
```

Hi @timgl ! Please see below:

```
Internal Server Error: /login
Traceback (most recent call last):
  File "/srv/posthog/.pyenv/versions/3.7.7/lib/python3.7/site-packages/django/core/handlers/exception.py", line 34, in inner
    response = get_response(request)
  File "/srv/posthog/.pyenv/versions/3.7.7/lib/python3.7/site-packages/django/core/handlers/base.py", line 115, in _get_response
    response = self.process_exception_by_middleware(e, request)
  File "/srv/posthog/.pyenv/versions/3.7.7/lib/python3.7/site-packages/django/core/handlers/base.py", line 113, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/srv/posthog/posthog/posthog/urls.py", line 39, in login_view
    posthoganalytics.capture(user.distinct_id, 'user logged in')
  File "/srv/posthog/.pyenv/versions/3.7.7/lib/python3.7/site-packages/posthoganalytics/__init__.py", line 45, in capture
    _proxy('capture', distinct_id=distinct_id, event=event, properties=properties, context=context, timestamp=timestamp, message_id=message_id)
  File "/srv/posthog/.pyenv/versions/3.7.7/lib/python3.7/site-packages/posthoganalytics/__init__.py", line 141, in _proxy
    fn(*args, **kwargs)
  File "/srv/posthog/.pyenv/versions/3.7.7/lib/python3.7/site-packages/posthoganalytics/client.py", line 94, in capture
    require('distinct_id', distinct_id, ID_TYPES)
  File "/srv/posthog/.pyenv/versions/3.7.7/lib/python3.7/site-packages/posthoganalytics/client.py", line 280, in require
    raise AssertionError(msg)
AssertionError: distinct_id must have (<class 'numbers.Number'>, (<class 'str'>,)), got: None
```

--- a/posthog/urls.py
+++ b/posthog/urls.py
@@ -36,7 +36,8 @@ def login_view(request):
         user = authenticate(request, email=email, password=password)
         if user is not None:
             login(request, user, backend='django.contrib.auth.backends.ModelBackend')
-            posthoganalytics.capture(user.distinct_id, 'user logged in')
+            if user.distinct_id:
+                posthoganalytics.capture(user.distinct_id, 'user logged in')
             return redirect('/')
         else:
             return render_template('login.html', request=request, context={'email': email, 'error': True})

